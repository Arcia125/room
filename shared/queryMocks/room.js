import * as generate from '../mockData/generate';
import { GET_ROOM } from '../../src/graphql/getRoom';
import { NEW_ROOM_MESSAGE } from '../../src/graphql/newRoomMessage';
import { users } from '../mockData/users';
import { NEW_ROOM_USER } from '../../src/graphql/newRoomUser';
import { JOIN_ROOM } from '../../src/graphql/joinRoom';
import { ADD_ROOM } from '../../src/graphql/addRoom';

const mockRoom = generate.room({
  id: 'test-id',
  name: 'test-room',
  messages: generate.messages(2),
  users,
});

const { id: roomId } = mockRoom;

const getRoomMock = {
  request: {
    query: GET_ROOM,
    variables: { id: roomId },
  },
  result: {
    data: {
      room: mockRoom,
    },
  },
};

const newRoomMessageMock = {
  request: {
    query: NEW_ROOM_MESSAGE,
    variables: { roomId },
  },
  result: {
    data: {
      newRoomMessage: generate.message(),
    },
  },
};

const newRoomUserMock = {
  request: {
    query: NEW_ROOM_USER,
    variables: { roomId },
  },
  result: {
    data: {
      newRoomUser: generate.user(),
    },
  },
};

const joinRoomMock = {
  request: {
    query: JOIN_ROOM,
    variables: { roomId },
  },
  result: {
    data: {
      joinRoom: {
        success: true,
        __typename: 'SuccessResult',
      },
    },
  },
};

const addRoomMock = {
  request: {
    query: ADD_ROOM,
  },
  result: {
    data: {
      addRoom: generate.room(),
    },
  },
};

export {
  mockRoom,
  getRoomMock,
  newRoomMessageMock,
  newRoomUserMock,
  joinRoomMock,
  addRoomMock,
};
