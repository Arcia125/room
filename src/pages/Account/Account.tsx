import React from 'react';
import PropTypes from 'prop-types';
import { GET_CURRENT_USER } from '../../graphql/getCurrentUser';
import { useQuery } from '@apollo/react-hooks';

const Account = () => {
  const getCurrentUserQuery = useQuery(GET_CURRENT_USER);

  if (getCurrentUserQuery.loading) return 'Loading...';
  if (getCurrentUserQuery.error) return getCurrentUserQuery.error.message;

  const currentUser =
    getCurrentUserQuery.data && getCurrentUserQuery.data.currentUser;
  return (
    <div>
      <h1>Account</h1>
      <label htmlFor="email">Email</label>
      <input id="email" readOnly value={currentUser.email} />
      <label htmlFor="password">Username</label>
      <input id="password" readOnly value={currentUser.username} />
    </div>
  );
};

Account.propTypes = {};

export default Account;
