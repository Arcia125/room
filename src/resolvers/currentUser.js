import { getUser, setUser, getToken } from '../utils/user';
// let user = {
//   id: null,
//   username: null,
//   __typename: 'User'
// };

// let user = getUser();

const currentUser = {
  defaults: {
    currentUser: getUser()
  },
  resolvers: {
    Mutation: {
      saveUser: (_, { username, id, email, token }, { cache }) => {
        const user = { id, username, email, __typename: 'User' };
        console.log('saveUser resolver', user, token);
        // console.log('saving user ', user);
        // cache.writeQuery({
        //   query: GET_CURRENT_USER,
        //   data: { currentUser: user },
        // });

        setUser({ user, cache, token });

        return { user, token, __typename: 'AuthPayload' };
        // cache.writeData({
        //   currentUser: {
        //     id,
        //     username,
        //     __typename: 'User',
        //   },
        // });
      }
    },
    Query: {
      currentUser: (root, args, { cache }, info) => {
        // debugger;
        console.log('getting current user');
        // const queryResult = cache.readQuery({
        //   query: GET_CURRENT_USER,
        // });
        // console.log('readQuery', queryResult);
        const user = getUser();
        return user ? user : null;
        // return {
        //   user: getUser(),
        //   token: getToken(),
        // };
      }
    }
  }
};

export { currentUser };
