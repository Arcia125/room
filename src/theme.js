const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px'
  }
};

export { theme };
