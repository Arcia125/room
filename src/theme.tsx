import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px',
  },
  colors: {
    gray: '#B3B2C0',
    grayDark: '#1f1f1f',
    lightBlue: '#AEAFE8',
    blue: '#AC2FC9',
    lightPurple: '#EFEEF8',
    purple: '#9056D4',
    pink: '#DD6589',
    bleh: '#8A4966',
  },
};

const Provider: React.FunctionComponent<{ children: ReactElement }> = ({
  children,
  ...restProps
}) => (
  <ThemeProvider theme={theme} {...restProps}>
    {children}
  </ThemeProvider>
);

export { theme, Provider };
