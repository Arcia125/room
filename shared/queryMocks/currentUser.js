import { users } from '../mockData/users';
import { GET_CURRENT_USER } from '../../src/graphql/getCurrentUser';
import { currentUser } from '../../src/resolvers/currentUser';


const mockUser = users[0];

const currentUserMock = {
  request: {
    query: GET_CURRENT_USER,
  },
  result: {
    data: {
      currentUser: mockUser
    }
  }
};

const currentUserMockResolvers = {
  Mutation: currentUser.resolvers.Mutation,
  Query: {
    currentUser: () => mockUser
  }
};

const nullCurrentUserMock = {
  request: {
    query: GET_CURRENT_USER,
  },
  result: {
    data: {
      currentUser: null
    }
  }
};

const nullCurrentUserMockResolvers = {
  Mutation: currentUser.resolvers.Mutation,
  Query: {
    currentUser: () => null
  }
};



export { currentUserMock, currentUserMockResolvers, nullCurrentUserMock, nullCurrentUserMockResolvers };