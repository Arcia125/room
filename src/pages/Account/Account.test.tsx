import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Account from '.';
import { Provider } from '../../theme';
import { currentUserMockResolvers } from '../../../shared/queryMocks/currentUser';

afterEach(cleanup);

describe('Account', () => {
  it('renders without crashing', async () => {
    const { container, asFragment } = render(
      <MockedProvider addTypename resolvers={currentUserMockResolvers}>
        <Provider>
          <Account />
        </Provider>
      </MockedProvider>
    );

    expect(container).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("finishes loading and displays the current user's username", async () => {
    const { getByDisplayValue, asFragment } = render(
      <MockedProvider addTypename resolvers={currentUserMockResolvers}>
        <Provider>
          <Account />
        </Provider>
      </MockedProvider>
    );

    await wait(() =>
      expect(
        getByDisplayValue(currentUserMockResolvers.Query.currentUser().username)
      ).toBeInTheDocument()
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
