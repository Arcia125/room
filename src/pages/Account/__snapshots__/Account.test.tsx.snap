// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account finishes loading and displays the current user's username 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Account
    </h1>
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email"
      readonly=""
      value="kevin@test.com"
    />
    <label
      for="password"
    >
      Username
    </label>
    <input
      id="password"
      readonly=""
      value="Kevin"
    />
  </div>
</DocumentFragment>
`;

exports[`Account renders without crashing 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
